fnl/spec/nfnl/fs_spec.fnl

```diff
@@ -20,7 +23,9 @@
   (fn []
     (it "returns the OS path separator (test assumes Linux)"
         (fn []
-          (assert.equals "/" (fs.path-sep))))))
+          (if (windows)
+              (assert.equals "\\" (fs.path-sep))
+              (assert.equals "/" (fs.path-sep)))))))
```

Comment:
So rather than a `windows` function and then many places having to branch, what I want to change this to is what we have in Aniseed.

https://github.com/Olical/aniseed/blob/7bc09736f3651c10d29b82d1a465b7f540614be1/fnl/aniseed/fs.fnl#L27-L34

So we'd have a function that tells us the path-sep for this OS then other path functions that build on this to abstract the idea away like we do in Conjure.

https://github.com/Olical/conjure/blob/a81977726f726b21d4bd75cea5bbbb7022005507/fnl/conjure/fs.fnl#L28-L42

I have a feeling _some_ of these ideas are already ported to nfnl but I would like to do it for more of them. Then the tests can be written in such a way that they never explicitly branch, they just run their paths through a function that standardises them for the OS you're on.

You don't have to do this, I'm happy to add on top of your PR and then hand it back to you with my changes for you to do further tweaks to if required. Or just to verify that you're happy with the behaviour.

Reply:
Ok I left the windows branch here but rewrote most of the tests to use correct-separators and standardize-path rather than branching. The places I left the conditional were for tests that were in some way directly about the separator.